src/zenml/artifact_stores/step_logging_utils.py

```diff
@@ -0,0 +1,105 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utilities for step logs."""
+
+import logging
+import os
+from typing import TYPE_CHECKING, Optional
+from uuid import UUID, uuid4
+
+from zenml.artifact_stores.base_artifact_store_logging_handler import (
+    ArtifactStoreLoggingHandler,
+)
+from zenml.io import fileio
+from zenml.logger import get_logger
+
+if TYPE_CHECKING:
+    from zenml.artifact_stores import BaseArtifactStore
+
+
+logger = get_logger(__name__)
+
+
+def prepare_logs_uri(
+    artifact_store: "BaseArtifactStore",
+    pipeline_run_id: UUID,
+    step_name: str,
+    log_key: Optional[str] = None,
+) -> str:
+    """Generates and prepares a URI for the log file for a step.
+
+    Args:
+        artifact_store: The artifact store on which the artifact will be stored.
+        pipeline_run_id: The id of the pipeline run.
+        step_name: Name of the step.
+        log_key: The unique identification key of the log file.
+
+    Returns:
+        The URI of the output artifact.
+    """
+    if log_key is None:
+        log_key = str(uuid4())
+
+    logs_base_uri = os.path.join(
```

Comment:
@schustmi I have a question about this part: Do you think this path for the logs make sense?

Reply:
@fa9r In this case I suppose a new uuid4 will be generated and another file made right?

Reply:
Yeah @schustmi can comment

Reply:
Actually now checking it, wouldn't it make more sense to have it in a similar place to the artifacts? Those are in `artifact_store.path/step_name/output_name/step_run_id`, maybe we could place the logs in `artifact_store.path/step_name/logs/step_run_id`?

Reply:
We dont have the `step_run_id` unfortuantely at the point we create this so i went with uuid generation approach but added the `/logs`. Check it out!

Reply:
Is this still an open question?